stac_api/runtime/src/app.py

```diff
@@ -62,10 +74,37 @@
         CORSMiddleware,
         allow_origins=api_settings.cors_origins,
         allow_credentials=True,
-        allow_methods=["GET", "POST", "OPTIONS"],
+        allow_methods=["GET", "POST", "PUT", "OPTIONS"],
         allow_headers=["*"],
     )
 
+# Require auth for all endpoints that create, modify or delete data.
+add_route_dependencies(
+    app.router.routes,
+    [
+        {"path": "/collections", "method": "POST", "type": "http"},
+        {"path": "/collections/{collectionId}", "method": "PUT", "type": "http"},
+        {"path": "/collections/{collectionId}", "method": "DELETE", "type": "http"},
+        {"path": "/collections/{collectionId}/items", "method": "POST", "type": "http"},
+        {
+            "path": "/collections/{collectionId}/items/{itemId}",
+            "method": "PUT",
+            "type": "http",
+        },
+        {
+            "path": "/collections/{collectionId}/items/{itemId}",
+            "method": "DELETE",
+            "type": "http",
+        },
+        {
+            "path": "/collections/{collectionId}/bulk_items",
+            "method": "POST",
+            "type": "http",
+        },
+    ],
+    [Depends(auth.validated_token)],
```

Comment:
```suggestion
    [Security(auth.validated_token)],
```

https://fastapi.tiangolo.com/reference/dependencies/#security